File: problems/atcoder/beginner_contest/176_20200822/e.py
h, w, m = map(int, input().split())

count_h = [0 for i in range(h)]
count_w = [0 for i in range(w)]
points = set()
for i in range(m):
    y, x = map(int, input().split())
    y -= 1
    x -= 1
    points.add((y, x))
    count_h[y] += 1
    count_w[x] += 1

ch_max, cw_max = max(count_h), max(count_w)
ys = [y for y in range(len(count_h)) if count_h[y] == ch_max]
xs = [x for x in range(len(count_w)) if count_w[x] == cw_max]

    
ans = ch_max + cw_max
if len(ys) * len(xs) <= m:
    def _find():
        for x in xs:
            for y in ys:
                if (y, x) not in points:
                    return True
        return False
    if not _find():
        ans -= 1
print(ans)

File: problems/atcoder/beginner_contest/176_20200822/e.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main() {
    int h, w, m;
    cin >> h >> w >> m;
    vector<int> count_h(h, 0);
    vector<int> count_w(w, 0);

    set<pair<int, int>> points;
    for (int i = 0; i < m; i++) {
        int y, x;
        cin >> y >> x;
        y--; x--;
        points.insert(make_pair(y, x));
        count_h[y]++; count_w[x]++;
    }
    int ch_max = *max_element(count_h.begin(), count_h.end());
    int cw_max = *max_element(count_w.begin(), count_w.end());
    int ans = 0;
    for (int y = 0; y < h; y++) {
        for (int x = 0; x < w; x++) {
            if (count_h[y] != ch_max || count_w[x] != cw_max) {
                continue;
            }
            if (points.find(make_pair(y, x)) != points.end()) {
                ans = max(ans, count_h[y] + count_w[x] - 1);
            } else {
                ans = max(ans, count_h[y] + count_w[x]);
            }
        }
    }
    cout << ans << endl;
    return 0;
}